docs/onboarding.md

```diff
@@ -0,0 +1,157 @@
+
+- If the instance has a `password` set, to the `/login` page
+- If the instance has a `registerToken` set, to the `onboarding` application.
+
+After login, the user is always redirected to the `onboarding` application. It is the `onboarding` application responsibility to check if registering is complete and reredirect to home.
+
+
+## Necessary new routes
+
+### GET /instance
+
+If the user is not logged in, display public instance information
+#### Request
+```http
+GET https://alice.example.com/instance
+```
+#### Response
+```json
+{
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+If the user is logged in, display all instance information, except password
+#### Request
+```
+GET https://alice.example.com/instance
+Cookie sessionid=xxxx
+```
+#### Response
+```json
+{
+  "_rev":"3-56521545485448482",
+  "domain": "alice.example.com",
+  "owner-email": "alice@example.com",
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+### POST /instance
+
+If the user is not logged in, allow to set the password
+
+#### Request
+```http
+POST https://alice.example.com/instance
+Content-type: application/x-www-form-urlencoded
+registerToken=4256595659594842&password=my-secret
+```
+
+#### Response
+```
+301 Redirect
+Set-Cookie: sessionid=xxxx .example.com
+Location: onboarding.alice.example.com
+```
+
+If the user is logged in, allow to set instance fields
+
+#### Request
+```http
+POST https://alice.example.com/instance
+Content-type: application/json
+Cookie sessionid=xxxxx
+Authorization: base64(onboarding:onboardingapptoken)
+```
+```json
+{
+  "_rev":"3-56521545485448482",
+  "tz": "Europe/Berlin"
+}
+```
+
+#### Response
+```
+200 OK
+Content-type: application/json
+Cookie sessionid=xxxxx
+Authorization: base64(onboarding:onboardingapptoken)
+```
+```json
+{
+  "_rev":"4-15454854484828845221285",
+  "tz": "Europe/Berlin",
+  "domain": "alice.example.com",
+  "owner-email": "alice@example.com",
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+## Flow Example
+
+- The server administrator Bob creates an instance through the CLI. He knows the instance should be in french for an user named `alice`.
+```
+cozy-stack instance add alice.example.com --locale fr --public-name "Alice Martin" --tz Europe/Paris
+>> https://alice.cozycloud.cc?registerToken=42456565213125454842
+```
+The instance is created
+```json
+{
+  "domain": "alice.example.com",
+  "publicName": "Alice Martin",
+  "tz": "Europe/Paris",
+  "locale":"fr"
+}
+```
+- Eve knows Alice just had an instance created, she goes to `https://alice.cozycloud.cc`. There is no `registerToken`, so she only see a message (in french) along the lines of "This is the cozy for Alice Martin, this register link is incorrect, if you are Alice Martin please ask your sysadmin for a new link".
```

Comment:
If I understand correctly, the registerToken will be required since the first onboarding steps here. Maybe we should be more explicit about that.